ruby/tests/basic.rb

```diff
@@ -1174,6 +1174,36 @@ def test_json
       Foo.encode_json(Foo.new(bar: bar, baz: [baz1, baz2]))
     end
 
+    def test_json_emit_defaults
+      # TODO: Fix JSON in JRuby version.
+      return if RUBY_PLATFORM == "java"
+      m = TestMessage.new
+
+      expected = '{"optionalInt32":0,"optionalInt64":0,"optionalUint32":0,"optionalUint64":0,"optionalBool":false,"optionalFloat":0,"optionalDouble":0,"optionalString":"","optionalBytes":"","optionalEnum":"Default","repeatedInt32":[],"repeatedInt64":[],"repeatedUint32":[],"repeatedUint64":[],"repeatedBool":[],"repeatedFloat":[],"repeatedDouble":[],"repeatedString":[],"repeatedBytes":[],"repeatedMsg":[],"repeatedEnum":[]}'
```

Comment:
This is a string, not a hash literal. There's not much freedom to format as it should match the output exactly. I don't really see any examples of large string literals in the file, except in `test_inspect`.

Should I...
a) Use `<<STR` and remove all whitespace.
b) Parse the JSON string and compare with a hash literal, which can be nicely formatted.